src/test/java/org/broadinstitute/consent/http/service/DatasetServiceTest.java

```diff
@@ -767,10 +768,46 @@ void testApproveDataset_DenyDataset() throws Exception {
     verify(emailService, times(1)).sendDatasetDeniedMessage(
         user,
         "DAC NAME",
-        "DUOS-000001"
+        "DUOS-000001",
+        "dacEmail@gmail.com"
     );
   }
 
+  @Test
+  void testApproveDataset_DenyDataset_WithNoDACEmail() throws Exception {
+    Dataset dataset = new Dataset();
+    dataset.setDataSetId(1);
+    User user = new User();
+    user.setUserId(1);
+    user.setEmail("asdf@gmail.com");
+    user.setDisplayName("John Doe");
+    Boolean payloadBool = false;
+    Dataset updatedDataset = new Dataset();
+    updatedDataset.setDataSetId(1);
+    updatedDataset.setDacApproval(payloadBool);
+
+    when(datasetDAO.findDatasetById(any())).thenReturn(updatedDataset);
+    dataset.setAlias(1);
+    dataset.setDacId(3);
+    Dac dac = new Dac();
+    dac.setName("DAC NAME");
+    initService();
+    when(dacDAO.findById(3)).thenReturn(dac);
+
+    Dataset returnedDataset = datasetService.approveDataset(dataset, user, payloadBool);
+    assertEquals(dataset.getDataSetId(), returnedDataset.getDataSetId());
+    assertFalse(returnedDataset.getDacApproval());
+
+    // do not send denied email
+    verify(emailService, times(0)).sendDatasetDeniedMessage(
```

Comment:
A nice alias for `times(0)` is `never()`
```suggestion
    verify(emailService, never()).sendDatasetDeniedMessage(
```

Reply:
Nice